src/mon/AuthMonitor.cc

```diff
@@ -1933,6 +1934,46 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
   return false;
 }
 
+
+int AuthMonitor::_is_perm_in_mdscaps_correct(const string& mdscaps,
+					     ostream& out)
+{
+  unsigned int num_of_caps = 1;
+  for (auto j : mdscaps) {
+    if (j == ',') {
+      ++num_of_caps;
+    }
+  }
+
+  unsigned int beg = 0;
+  unsigned int len = num_of_caps ? mdscaps.size() : mdscaps.find(",");
+  for (unsigned int i = 0; i != num_of_caps; ++i) {
+    if (i > 0) {
+      beg = mdscaps.find(",", beg) + 1;
+      // commas may or may not be followed by a space
+      // "allow rw fsname=cephfs1, allow rw fsname=cephfs2" vs.
+      // "allow rw fsname=cephfs,allow rw fsname=cephfs2"
+      if (mdscaps[beg] == ' ') {
+	++beg;
+      }
+
+      if (i < num_of_caps) {
+	len = mdscaps.find(",", beg) - beg;
+      } else {
```

Comment:
We are not going to reach this else condition as its already taken care of in the `for` loop and I don't see any  increments for `i` other than the `for` loop which can lead to this else condition.

Reply:
True.